### PR TITLE
Update Ansible and fix E2E tests

### PR DESCRIPTION
**Pull request by CyanDevs**
Changes to Ansible:
* Update to Ansible 4.10 for Ubuntu 18.04
* Update to Ansible 5.2.0 for Ubuntu 20.04
* Added Ansible Galaxy modules that cannot be assumed are built-in to Ansible installation script
* Fixed Ansible linting issues and added an .ansible_lint file to skip certain rules 503 (tasks vs handlers), and 204 (line length)
* Fixed Intel PSW/DCAP version for Ansible Windows role
* Install Docker Python library when needed for Ansible setup
* Use Docker modules for Ansible instead of running bash commands
* Install docker-ce-cli and containerd.io when installing docker-ce as recommended by Docker's installation documentation.

* Update README.md for Ansible to latest supported OSes

Other changes
* Remove dependency on Docker oetools-deploy for packer builds, as the image is no longer maintained and the packer version within no longer works for our builds.
* Fix broken references to Jenkins job paths
* Fix link to devkits url
* Add `common.installAzureCLI` to OpenEnclaveJenkinsLibrary for future use

### CODE REVIEW

File: scripts/ansible/oe-windows-acc-setup.yml [hunk 1]
```diff
@@ -8,10 +8,10 @@
   tasks:
     - name: OE setup | Set installer URLs from the OE storage account
       set_fact:
-        intel_psw_url:       "https://registrationcenter-download.intel.com/akdlm/irc_nas/18233/Intel%20SGX%20PSW%20for%20Windows%20v2.14.100.3.exe"
-        intel_psw_hash:      "6DA4DB8E0495404253FA9CF737622A5AE8F4943F611180EF12FB48C0EAA13812"
-        intel_dcap_url:      "https://registrationcenter-download.intel.com/akdlm/irc_nas/18232/Intel%20SGX%20DCAP%20for%20Windows%20v1.12.100.3.exe"
-        intel_dcap_hash:     "885772E40F66DE2D2E2AE3E214AEB04E51F8173C8F61F013C1DAD57C01D846A5"
+        intel_dcap_url:      "https://registrationcenter-download.intel.com/akdlm/irc_nas/18336/Intel%20SGX%20DCAP%20for%20Windows%20v1.12.101.1.exe"
```
Comment (CyanDevs) [inline scripts/ansible/oe-windows-acc-setup.yml:11]: Nit: Undo line switcharoo

File: scripts/ansible/requirements.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 cryptography==3.3.2
-ansible==2.10.7
+ansible==5.2.0
```
Comment (CyanDevs) [inline scripts/ansible/requirements.txt:4]: Ubuntu 18.04 only has Ansible 5.0.0 latest on pip.
```
[](https://oe-jenkins-dev.westeurope.cloudapp.azure.com/blue/organizations/jenkins/Private%2FCICD_Infrastructure%2FLinux-Docker-Container-Build/detail/Linux-Docker-Container-Build/67/pipeline/#step-125-log-1593)[](https://oe-jenkins-dev.westeurope.cloudapp.azure.com/blue/organizations/jenkins/Private%2FCICD_Infrastructure%2FLinux-Docker-Container-Build/detail/Linux-Docker-Container-Build/67/pipeline/#step-125-log-1594)Collecting ansible==5.2.0 (from -r /ansible/requirements.txt (line 4))

  Could not find a version that satisfies the requirement ansible==5.2.0 (from -r /ansible/requirements.txt (line 4)) (from versions: 1.0, 1.1, 1.2, 1.2.1, 1.2.2, 1.2.3, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.4, 1.4.1, 1.4.2, 1.4.3, 1.4.4, 1.4.5, 1.5, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.5.5, 1.6, 1.6.1, 1.6.2, 1.6.3, 1.6.4, 1.6.5, 1.6.6, 1.6.7, 1.6.8, 1.6.9, 1.6.10, 1.7, 1.7.1, 1.7.2, 1.8, 1.8.1, 1.8.2, 1.8.3, 1.8.4, 1.9.0.1, 1.9.1, 1.9.2, 1.9.3, 1.9.4, 1.9.5, 1.9.6, 2.0.0.0, 2.0.0.1, 2.0.0.2, 2.0.1.0, 2.0.2.0, 2.1.0.0, 2.1.1.0, 2.1.2.0, 2.1.3.0, 2.1.4.0, 2.1.5.0, 2.1.6.0, 2.2.0.0, 2.2.1.0, 2.2.2.0, 2.2.3.0, 2.3.0.0, 2.3.1.0, 2.3.2.0, 2.3.3.0, 2.4.0.0, 2.4.1.0, 2.4.2.0, 2.4.3.0, 2.4.4.0, 2.4.5.0, 2.4.6.0, 2.5.0a1, 2.5.0b1, 2.5.0b2, 2.5.0rc1, 2.5.0rc2, 2.5.0rc3, 2.5.0, 2.5.1, 2.5.2, 2.5.3, 2.5.4, 2.5.5, 2.5.6, 2.5.7, 2.5.8, 2.5.9, 2.5.10, 2.5.11, 2.5.12, 2.5.13, 2.5.14, 2.5.15, 2.6.0a1, 2.6.0a2, 2.6.0rc1, 2.6.0rc2, 2.6.0rc3, 2.6.0rc4, 2.6.0rc5, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.6.6, 2.6.7, 2.6.8, 2.6.9, 2.6.10, 2.6.11, 2.6.12, 2.6.13, 2.6.14, 2.6.15, 2.6.16, 2.6.17, 2.6.18, 2.6.19, 2.6.20, 2.7.0.dev0, 2.7.0a1, 2.7.0b1, 2.7.0rc1, 2.7.0rc2, 2.7.0rc3, 2.7.0rc4, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.7.5, 2.7.6, 2.7.7, 2.7.8, 2.7.9, 2.7.10, 2.7.11, 2.7.12, 2.7.13, 2.7.14, 2.7.15, 2.7.16, 2.7.17, 2.7.18, 2.8.0a1, 2.8.0b1, 2.8.0rc1, 2.8.0rc2, 2.8.0rc3, 2.8.0, 2.8.1, 2.8.2, 2.8.3, 2.8.4, 2.8.5, 2.8.6, 2.8.7, 2.8.8, 2.8.9, 2.8.10, 2.8.11, 2.8.12, 2.8.13, 2.8.14, 2.8.15, 2.8.16rc1, 2.8.16, 2.8.17rc1, 2.8.17, 2.8.18rc1, 2.8.18, 2.8.19rc1, 2.8.19, 2.8.20rc1, 2.8.20, 2.9.0b1, 2.9.0rc1, 2.9.0rc2, 2.9.0rc3, 2.9.0rc4, 2.9.0rc5, 2.9.0, 2.9.1, 2.9.2, 2.9.3, 2.9.4, 2.9.5, 2.9.6, 2.9.7, 2.9.8, 2.9.9, 2.9.10, 2.9.11, 2.9.12, 2.9.13, 2.9.14rc1, 2.9.14, 2.9.15rc1, 2.9.15, 2.9.16rc1, 2.9.16, 2.9.17rc1, 2.9.17, 2.9.18rc1, 2.9.18, 2.9.19rc1, 2.9.19, 2.9.20rc1, 2.9.20, 2.9.21rc1, 2.9.21, 2.9.22rc1, 2.9.22, 2.9.23rc1, 2.9.23, 2.9.24rc1, 2.9.24, 2.9.25rc1, 2.9.25, 2.9.26rc1, 2.9.26, 2.9.27rc1, 2.9.27, 2.10.0a1, 2.10.0a2, 2.10.0a3, 2.10.0a4, 2.10.0a5, 2.10.0a6, 2.10.0a7, 2.10.0a8, 2.10.0a9, 2.10.0b1, 2.10.0b2, 2.10.0rc1, 2.10.0, 2.10.1, 2.10.2, 2.10.3, 2.10.4, 2.10.5, 2.10.6, 2.10.7, 3.0.0b1, 3.0.0rc1, 3.0.0, 3.1.0, 3.2.0, 3.3.0, 3.4.0, 4.0.0a1, 4.0.0a2, 4.0.0a3, 4.0.0a4, 4.0.0b1, 4.0.0b2, 4.0.0rc1, 4.0.0, 4.1.0, 4.2.0, 4.3.0, 4.4.0, 4.5.0, 4.6.0, 4.7.0, 4.8.0, 4.9.0, 4.10.0, 5.0.0a1, 5.0.0a2, 5.0.0a3, 5.0.0b1, 5.0.0b2, 5.0.0rc1)

No matching distribution found for ansible==5.2.0 (from -r /ansible/requirements.txt (line 4))
```

Comment (CyanDevs) [reply scripts/ansible/requirements.txt]: Ansible 5 requires Python 3.8, and Ubuntu only has Python 3.6 by default. While we could install that and use a virtualenv, there is no perceived functional difference for us between Ansible 4.10 and 5 so it is best to keep things simple until there is a need to move to Ansible 5 on Ubuntu 18.04.

So, the requirement is now Ansible >= 4.10 or <= 5.2.0

**Comment by CyanDevs** [comment]
Successful E2E test with https://oe-jenkins-dev.westeurope.cloudapp.azure.com/job/OpenEnclave/job/E2E%20Testing/38/ for https://github.com/openenclave/openenclave/commit/3dcc03748601c14f69b8f4ca6229788ec8dbdd59
Waiting E2E test results for the most recent commit.

File: .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile [hunk 3]
```diff
@@ -275,6 +284,34 @@ node(params.AGENTS_LABEL) {
...
+                sudo apt-get update
+                sudo apt-get -y install azure-cli
+            """
+        }
+    }
+    stage("Install Ansible") {
+        retry(10) {
+            sh """#!/bin/bash
+                ${helpers.WaitForAptLock()}
+                sudo whoami
```
Comment (rs--) [inline .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile:310]: is this command needed?

File: .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile [hunk 4]
```diff
@@ -275,6 +284,34 @@ node(params.AGENTS_LABEL) {
 
         println("IMAGE_VERSION: ${image_version}\nIMAGE_ID: ${image_id}")
     }
+    stage("Install Azure CLI") {
+        retry(10) {
+            sh """
+                sleep 5
```
Comment (rs--) [inline .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile:290]: Is it possible to call `installAzureCLI()` here instead?

**Review by rs-- — APPROVED** [review]

File: .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile (hunk 4 above)
Comment (CyanDevs) [reply .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile:290]: With how OpenEnclaveJenkinsLibrary defaults to master, it is not possible until `common.installAzureCLI()` (this PR) is merged into master.

File: .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile (hunk 3 above)
Comment (CyanDevs) [reply .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile:310]: It is not, thank you :)

File: .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile [hunk 5]
```diff
@@ -12,6 +12,7 @@ GLOBAL_TIMEOUT_MINUTES = 480
 JENKINS_USER_CREDS_ID = "oeadmin-credentials"
 OETOOLS_REPO = "oejenkinscidockerregistry.azurecr.io"
 OETOOLS_REPO_CREDENTIALS_ID = "oejenkinscidockerregistry"
+SERVICE_PRINCIPAL_CREDENTIALS_ID = 'SERVICE_PRINCIPAL_OSTCLAB'
```
[the remark this answers is not in the document]
Comment (CyanDevs) [reply .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile:15]: Technically as no interpolation is required the single quotes are preferred. Will change them all to single quotes.

**Comment by CyanDevs** [comment]
E2E tests for the latest commit: https://oe-jenkins-dev.westeurope.cloudapp.azure.com/job/OpenEnclave/job/E2E%20Testing/43/

**Review by radhikaj — APPROVED** [review]
bors r+

**Pull request merged**